libraries/provision/install_sync_gateway.py

```diff
@@ -152,8 +162,27 @@ def install_sync_gateway(cluster_config, sync_gateway_config, sg_ce=False,
         else:
             num_replicas = get_sg_replicas(cluster_config)
             playbook_vars["num_index_replicas"] = '"num_index_replicas": {},'.format(num_replicas)
+
+        if is_x509_auth(cluster_config):
+            playbook_vars[
+                "certpath"] = '"certpath": "/home/sync_gateway/certs/chain.pem",'
+            playbook_vars[
+                "keypath"] = '"keypath": "/home/sync_gateway/certs/pkey.key",'
+            playbook_vars[
+                "cacertpath"] = '"cacertpath": "/home/sync_gateway/certs/ca.pem",'
+            playbook_vars["server_scheme"] = "couchbases"
+            playbook_vars["server_port"] = ""
+            playbook_vars["x509_auth"] = True
+            generate_x509_certs(cluster_config, bucket_names)
```

Comment:
everytime we run test, it will generate x509 certs?

Reply:
Yes, certs are needed for different configs and different username. Certs will change for any change in required parameters, like bucket, user etc.